### PR TITLE
Add "Maintaining state that could be derived state"

### PR DESCRIPTION
**Pull request by RicheyRyan**
Hey, so I'm finally getting some time for this and I figured I'd contribute to the mistakes and the glossary. My plan is to contribute bit by bit when I have a spare moment at work. I don't get into adding any particular structure as suggested in #13    . I reckon we can start worrying about that when we have a number of items. 

I wanted to give a real-world example without getting too much into the nitty-gritty of a particular scenario. Let me know what you think could be improved.

### CODE REVIEW

File: website/docs/common-mistakes/mistake1.md [hunk 1]
```diff
@@ -4,3 +4,46 @@ title: Common Mistakes
...
+
+```js
+const [dateOfBirth, setDateOBirth] = useState(null);
+const ageInYears = getAgeInYears(dateOfBirth);
+const daysUntilBirthday = getDaysUntilBirthday(dateOfBirth);
+```
+
+We can call `getAgeInYears` and `getDaysUntilBirthday` on each render. We know we will have the latest value for `dateOfBirth`, removing the need for the `useEffect`. We avoid any potential bugs related by deriving the state directly. We also don't need to write any synchronization code.
+
+You might be wondering, "What if it's very costly to calculate these derived state values? Won't this cause performance issues in my component?" which is a valid concern. However, in most cases, the values can be derived quickly, which won't impact performance. If you know a calculation will be costly or see performance issues; you can use the `useMemo` hook to run the calculations only when `dateOfBirth` updates.
```
Comment (srmagura) [inline website/docs/common-mistakes/mistake1.md:32]:
```suggestion
You might be wondering, "What if it's very costly to calculate these derived state values? Won't this cause performance issues in my component?" which is a valid concern. However, in most cases, the values can be derived quickly, which won't impact performance. If you know a calculation will be costly or see performance issues, you can use the `useMemo` hook to run the calculations only when `dateOfBirth` updates.
```

Comment (srmagura) [reply website/docs/common-mistakes/mistake1.md:32]: ^ Me caring too much about grammar 😂

**Review by srmagura — APPROVED** [review]
Awesome work, I love it!

**Comment by RicheyRyan** [comment]
I realized that the mistakes were on `mistakes-1`. I reckon we should go with a single top-level page at `/common-mistakes` for now and figure out how to break it down later. If you prefer to do something different, I can either do it now or revert. Once this is done I plan to just open a stream of additional PRs. I have a couple items queued in Grammarly already 🙂

**Comment by markerikson** [comment]
Sure, let's get this in.  I'll be honest and say I'm not sure this is exactly a match for the kind of content I've been picturing in my head (maybe I mentioned something like this in the initial thread?  can't remember where "common mistakes" came in.  oh, I see, probably here: https://github.com/markerikson/react-community-tools-practices-cheatsheet/discussions/1#discussioncomment-414586 )

But the initial entry is well written and this absolutely _is_ a common problem, so let's see what else makes sense to add here.

**Pull request merged**